### ISSUE TITLE
qemu: update to 6.0

### ISSUE THREAD

**Issue by X547** [haikuports/haikuports#5859]
QEmu 6.0 has [important change](https://github.com/qemu/qemu/commit/c346749ee9d75fcb11bb816d0665ce174425d667) that allow to use `-device ramfb` on RISC-V target (needed for Haiku RISC-V port).

Not urgent because I use TinyEMU for now.

**Comment by diversys** [comment]
qemu 6.0 switched to using SDL2, however to work properly it needs your SDL fixes (which @threedeyes adapted to current SDL2 version) which in turn needs mesa 21.
In a nutshell #5792 needs to be merged first.

**Comment by X547** [comment]
I have working Qemu 6.0 build. Important changes from upstream:
1. `-fPIE` must be disabled.
2. Path “/etc/resolv.conf” should be replaced with “/boot/system/settings/network/resolv.conf” in Slirp library (used to get DNS IP address).

Also I made RISC-V HTIF available for "virt" machine at address 0x40008000, same as in TinyEMU.

**Comment by rjzak** [comment]
2. Merged into upstream libslirp https://gitlab.freedesktop.org/slirp/libslirp/-/merge_requests/99

**Comment by davidkaroly** [comment]
Perhaps we can enable GTK frontend for qemu? At least at a first attempt it looks promising to me.

**Comment by diversys** [comment]
I've heard rumours that @threedeyes wanted to write a native frontend quite some time ago. 
Until then we can enable GTK frontend indeed.

**Comment by pulkomandy** [comment]
We used to have a native frontend written by mmlr. Let me guess: once again something that was removed from the patchsets because the upstream code compiles and runs?

**Comment by davidkaroly** [comment]
This one? https://github.com/mmlr/qemu-haiku
Latest version seems to be qemu-2.1.2 :disappointed:

**Comment by pulkomandy** [comment]
Yes, I think that's it. Did the user interface code change a lot since then? I don't know

**Comment by davidkaroly** [comment]
Yes I think the code has changed significantly. For now we can use 6.2.0 with GTK frontend, see #7607
Of course eventually a native frontend would be much better.

**Comment by korli** [comment]
Can this be closed?

**Comment by davidkaroly** [comment]
> Can this be closed?

I think so.
Recipe for qemu-6.2.0 is available in haikuports repo - even though marked as untested
Recipe for qemu-7.2.0 is available in PR #7789
